packages/grafana-data/src/types/datasource.ts

```diff
@@ -145,6 +145,11 @@ interface PluginMetaQueryOptions {
   maxDataPoints?: boolean;
   minInterval?: boolean;
 }
+interface PluginQueryCachingConfig {
+  enabled?: boolean;
```

Comment:
Okay, will move it

Reply:
Moved to `DataSourcePluginMeta`

Reply:
Sorry I was thinking adding `meta.queryOptions?.queryCachingTTL` as a bool to denote if the option should be visible/enabled. Looking at the changes now not sure these changes make sense so maybe revert https://github.com/grafana/grafana/pull/61968/commits/395c74b585c88356cf8c49400d6b5a64cb033fd4. Sorry

Reply:
Ah, makes sense.

On one hand, the field seems a little bit out of place on the top-level Datasource object. PluginMeta seems like a logic place to have something like this. On the other hand, mixing things from plugin.json and enterprise may cause issues later on.
 
Perhaps a frontend team should let me know if the original approach was good before I revert. I will reach out again today.

Reply:
I just want to verify that I understood how this works.

The values passed in cache config is default value + whether or not if it should be visible. Once you have saved the panel model the value store there will be used, correct?

I would vote to have it on the instance settings instead of in the meta. So I guess I'm in favor of reverting the commit.